### PR TITLE
Feat/notebook apis

### PR DESCRIPTION
**Pull request by chandreshkkhatri**
- notebook routes
- notebook model
- notebook validations

### CODE REVIEW

File: routes/notebooks.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        const description = req.body.description ?? originalNotebook.description;
+
+        const result = await Notebook.updateOne(
+            { _id: notebookId, ownerId: userId },
+            { $set: { name, description } }
+        );
+        res.status(200).send({ result });
+    }
+);
+
```
Comment (shital01) [inline routes/notebooks.js:126]: do we need to set updatedtimestamp in trash and restore or is it automatically happening

File: routes/notebooks.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                        notebook
+                    })
+                }
+            }
+        }
+
+        res.status(200).send({ savedEntries, unsavedEntries });
+    }
+);
+
```
Comment (shital01) [inline routes/notebooks.js:104]: in all of below apis findone and updateone what if there is no result is found(which wont be happeneing ) still handle that sceanario as well or check if it is already handled by eror middlewares

File: routes/notebooks.js (hunk 1 above)
Comment (chandreshkkhatri) [reply routes/notebooks.js:126]: not being handled automatically, have updated code to update the timestamp manually.

File: routes/notebooks.js (hunk 2 above)
Comment (chandreshkkhatri) [reply routes/notebooks.js:104]: have added checks to take care of these scenarios

**Pull request merged**